pkg/stanza/operator/parser/syslog/syslog.go

```diff
@@ -93,28 +111,52 @@ func (c Config) Build(logger *zap.SugaredLogger) (operator.Operator, error) {
 	}
 
 	return &Parser{
-		ParserOperator: parserOperator,
-		protocol:       c.Protocol,
-		location:       location,
+		ParserOperator:               parserOperator,
+		protocol:                     c.Protocol,
+		location:                     location,
+		enableOctetCounting:          c.EnableOctetCounting,
+		nonTransparentFramingTrailer: c.NonTransparentFramingTrailer,
 	}, nil
 }
 
-func buildMachine(protocol string, location *time.Location) (sl.Machine, error) {
-	switch protocol {
+// parseFunc a parseFunc determines how the raw input is to be parsed into a syslog message
+type parseFunc func(input []byte) (sl.Message, error)
+
+func (s *Parser) buildParseFunc() (parseFunc, error) {
+	switch s.protocol {
 	case RFC3164:
-		return rfc3164.NewMachine(rfc3164.WithLocaleTimezone(location)), nil
+		return func(input []byte) (sl.Message, error) {
+			machine := rfc3164.NewMachine(rfc3164.WithLocaleTimezone(s.location))
+			return machine.Parse(input)
```

Comment:
nit, any reason not to do this?
```suggestion
			return rfc3164.NewMachine(rfc3164.WithLocaleTimezone(s.location)).Parse(input)
```

Reply:
Ah yeah good call. This got lost in a refactor. I original had more logic in between these lines. I'll clean it up.